_posts/2018-02-23-save-memory-with-cursors.md

```diff
@@ -0,0 +1,95 @@
+---
+layout: post
+title: Save memory with cursors
+tags: PostgreSQL
+---
+
+Has your application ever run out of memory when running a query that returns a
+a significant amount of data?
+
+For example, in PHP, I have seen a script like the following die because it ate
+up too much memory on my application server.
+
+```php
+$sql = <<<SQL
+SELECT *
+FROM really_large_table
+SQL;
+
+$res = pg_query($sql);
+while ($row = pg_fetch_row($res)) {
+  // Process $row...
+}
+```
+
+So how can you reduce your application's memory usage when querying a large
+amount of data from Postgres?
+
+Well, your application is running out of memory because you are trying to query
+all of the data at once, and your application server cannot (or will not) hold
+that many results in memory—this is to be expected.
```

Comment:
You might want to rephrase this to make it clear that `pg_query()` is actually loading all of the data in memory. This might not be clear to people who have not yet experienced and debugged this, which I think is a big part of what you are wanting to convey 🐤